homeassistant/components/local_calendar/calendar.py

```diff
@@ -0,0 +1,154 @@
+"""Calendar platform for a Local Calendar."""
+
+from __future__ import annotations
+
+from datetime import datetime
+import logging
+from typing import Any
+
+from ical.calendar import Calendar
+from ical.calendar_stream import IcsCalendarStream
+from ical.event import Event
+from ical.store import EventStore
+from ical.types import Range, Recur
+from ical.util import use_local_timezone
+
+from homeassistant.components.calendar import (
+    ENTITY_ID_FORMAT,
+    EVENT_DESCRIPTION,
+    EVENT_END,
+    EVENT_RRULE,
+    EVENT_START,
+    EVENT_SUMMARY,
+    EVENT_UID,
+    CalendarEntity,
+    CalendarEntityFeature,
+    CalendarEvent,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import generate_entity_id
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_CALENDAR_NAME, DOMAIN
+from .store import LocalCalendarStore
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the local calendar platform."""
+    store = hass.data[DOMAIN][config_entry.entry_id]
+    ics = await store.async_load()
+    calendar = IcsCalendarStream.calendar_from_ics(ics)
+
+    name = config_entry.data[CONF_CALENDAR_NAME]
+    entity_id = generate_entity_id(ENTITY_ID_FORMAT, name, hass=hass)
+    entity = LocalCalendarEntity(
+        store, calendar, name, entity_id, unique_id=config_entry.entry_id
+    )
+    async_add_entities([entity], True)
+
+
+class LocalCalendarEntity(CalendarEntity):
+    """A calendar entity backed by a local iCalendar file."""
+
+    _attr_has_entity_name = True
+    _attr_supported_features = CalendarEntityFeature.MUTABLE
+
+    def __init__(
+        self,
+        store: LocalCalendarStore,
+        calendar: Calendar,
+        name: str,
+        entity_id: str,
+        unique_id: str,
+    ) -> None:
+        """Initialize LocalCalendarEntity."""
+        self._store = store
+        self._calendar = calendar
+        self._event: CalendarEvent | None = None
+        self._attr_name = name.capitalize()
+        self.entity_id = entity_id
+        self._attr_unique_id = unique_id
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+        return self._event
+
+    async def async_get_events(
+        self, hass: HomeAssistant, start_date: datetime, end_date: datetime
+    ) -> list[CalendarEvent]:
+        """Get all events in a specific time frame."""
+        with use_local_timezone(dt_util.DEFAULT_TIME_ZONE):
```

Comment:
What does this context manager do?

Reply:
From https://allenporter.github.io/ical/ical/util.html#use_local_timezone
> Set the local timezone to use when converting a date to datetime.
> This is expected to be used as a context manager when the default timezone used by python is not the timezone to be used for calendar operations (the attendees local timezone).

Example:
```
import datetime
import zoneinfo
from [ical.calendar](https://allenporter.github.io/ical/ical/calendar.html) import Calendar
from [ical.event](https://allenporter.github.io/ical/ical/event.html) import Event
from [ical.util](https://allenporter.github.io/ical/ical/util.html) import use_local_timezone

cal = Calendar()
cal.events.append(
    Event(
        summary="Example",
        start=datetime.date(2022, 2, 1),
        end=datetime.date(2022, 2, 2)
    )
)
# Use UTC-8 as local timezone
with use_local_timezone(zoneinfo.ZoneInfo("America/Los_Angeles")):
    # Returns event above
    events = cal.timeline.start_after(
        datetime.datetime(2022, 2, 1, 7, 59, 59, tzinfo=datetime.timezone.utc))

    # Does not return event above
    events = cal.timeline.start_after(
        datetime.datetime(2022, 2, 1, 8, 00, 00, tzinfo=datetime.timezone.utc))
```

My impression is that home assistant's timezone is not the same thing as the local time used by python. This API is trying to solve the problem of passing timezone information all the way down into a python library when the timezone comparison is being made. 

An alternative can be to pass it into the `timeline` object and have it do what it wants with this timezone from the caller. I'm happy to flex on how this works.  Also, this is where the current bug in CI is that i need to dig in and investigate.

Reply:
I'd require the date times passed by the caller to be timezone aware and just go with those.

Reply:
Done.